types/index.test-d.ts

```diff
@@ -1,36 +1,41 @@
-import fastify, { FastifyInstance, FastifyPluginAsync, FastifyRequest, FastifyReply } from 'fastify'
-import { Server } from 'http'
+import fastify, { FastifyInstance, FastifyRequest, FastifyReply } from 'fastify'
 import { Stats } from 'fs'
 import { expectAssignable, expectError, expectType } from 'tsd'
 import * as fastifyStaticStar from '..'
 import fastifyStatic, {
+  FastifyStaticPlugin,
   FastifyStaticOptions,
-  fastifyStatic as fastifyStaticNamed
+  fastifyStatic as fastifyStaticNamed,
+  FastifyStaticPluginDecorators
 } from '..'
+// TODO: remove after we land in fastify-plugin
+import { createPlugin } from 'fastify-plugin'
 
 import fastifyStaticCjsImport = require('..')
 const fastifyStaticCjs = require('..')
 
 const app: FastifyInstance = fastify()
 
-app.register(fastifyStatic)
-app.register(fastifyStaticNamed)
-app.register(fastifyStaticCjs)
-app.register(fastifyStaticCjsImport.default)
-app.register(fastifyStaticCjsImport.fastifyStatic)
-app.register(fastifyStaticStar.default)
-app.register(fastifyStaticStar.fastifyStatic)
-
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(fastifyStatic)
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(fastifyStaticNamed)
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(fastifyStaticCjsImport.default)
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(fastifyStaticCjsImport.fastifyStatic)
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(fastifyStaticStar.default)
-expectType<FastifyPluginAsync<FastifyStaticOptions, Server>>(
-  fastifyStaticStar.fastifyStatic
-)
+app.register(fastifyStatic, { root: '/' })
```

Comment:
If I properly understand the change, we now require an object to be always passed when we register a fastify plugin (so it's no longer optional). This is a breaking change, we should really careful if we want to ship it as it is, since it will break existing TS codebases

Reply:
An object is not necessary, fastify static plugin has a mandatory root param. Removing the object leads to

```
  Overload 1 of 3, (plugin: FastifyPluginCallback<FastifyPluginOptions, BaseFastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProviderDefault, FastifyDecorators>>, opts?: FastifyRegisterOptions<...> | undefined): BaseFastifyInstance<...> & ... 1 more ... & { ...; }, gave the following error.
    Argument of type FastifyStaticPlugin<BaseFastifyInstance<any, any, any, any, any, FastifyDecorators>> is not assignable to parameter of type FastifyPluginCallback<FastifyPluginOptions, BaseFastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProviderDefault, FastifyDecorators>>.
      Types of parameters opts and opts are incompatible.
        Property root is missing in type FastifyPluginOptions but required in type FastifyStaticOptions.
  Overload 2 of 3, (plugin: FastifyPluginAsync<FastifyPluginOptions, BaseFastifyInstance<RawServerDefault, IncomingMessage, ServerResponse<IncomingMessage>, FastifyBaseLogger, FastifyTypeProviderDefault, FastifyDecorators>>, opts?: FastifyRegisterOptions<...> | undefined): BaseFastifyInstance<...> & ... 1 more ... & { ...; }, gave the following error.
```

Making root optional fixes the error.

If anything, the possibility of not providing root was a bug as far as i can tell.